website/content/docs/connect/gateways/mesh-gateway/service-to-service-traffic-partitions.mdx

```diff
@@ -0,0 +1,172 @@
+
+### Modes
+
+Each upstream associated with a service mesh proxy can be configured so that it is routed through a mesh gateway.  
+Depending on your network, the proxy's connection to the gateway can operate in one of the following modes:
+
+* `none` - (Default) No gateway is used and a service mesh connect proxy makes its outbound connections directly
+  to the destination services.
+
+* `local` - The service mesh connect proxy makes an outbound connection to a gateway running in the same datacenter. The gateway at the outbound connection is responsible for ensuring that the data is forwarded to gateways in the destination partition. 
+
+* `remote` - The service mesh connect proxy makes an outbound connection to a gateway running in the destination datacenter. 
+  The gateway forwards the data to the final destination service.
+
+### Connect Proxy Configuration
+
+Set the proxy to the preferred [mode](#modes) to configure the service mesh proxy. You can specify the mode globally or within child configurations to control proxy behaviors at a lower level. Consul recognizes the following order of precedence if the gateway mode is configured in multiple locations the order of precedence:
+
+1. Upstream definition (highest priority)
+2. Service instance definition
+3. Centralized `service-defaults` configuration entry
+4. Centralized `proxy-defaults` configuration entry
+
+## Example Configurations
+
+Use the following example configurations to help you understand some of the common scenarios.
+
+### Enabling Gateways Globally
+
+The following `proxy-defaults` configuration will enable gateways for all Connect services in the `local` mode.
+
+```hcl
+Kind = "proxy-defaults"
+Name = "global"
+MeshGateway {
+   Mode = "local"
+}
+```
+
+### Enabling Gateways Per-Service
+
+The following `service-defaults` configuration will enable gateways for all Connect services with the name `web`.
+
+```hcl
+Kind = "service-defaults"
+Name = "web"
+MeshGateway {
+   Mode = "local"
+}
+```
+
+### Enabling Gateways for a Service Instance
+
+The following [Proxy Service Registration](/docs/connect/registration/service-registration)
+definition will enable gateways for `web` service instances in the `finance` partition.
+
+```hcl
+service {
+   name = "web-sidecar-proxy"
+   kind = "connect-proxy"
+   port = 8181
+   proxy {
+      destination_service_name = "web"
+      mesh_gateway {
+         mode = "local"
+      }
+      upstreams = [
+         {
+            destination_partition = "finance"
+            destination_namespace = "default"
+            destination_type = "service"
+            destination_name = "billing"
+            local_bind_port = 9090
+         }
+      ]
+   }
+}
+```
+
```

Comment:
We should showcase how to do this with k8s annotations as well.  Below is an example NOT using trproxy.
postgress is the upstream service located in another partitions.  Double check me, but I belive the syntax is `service_name.[partition_name].[namespace]:portNumber

```
consul.hashicorp.com/connect-inject: "true"
consul.hashicorp.com/connect-service-upstreams: "postgres.default.default:5432"
```